templates/theories/linear_algebra.md

```diff
@@ -0,0 +1,106 @@
+# Maths in lean : linear algebra
+
+### semimodules, modules and vector spaces ###
+
+#### algebra.module ####
+
+This file defines the typeclass `semimodule R M`, which gives an `R`-semimodule structure on the type `M`, and similarly `module R M` and `vector_space R M`.
+An additive commutative monoid `M` is a semimodule over the semiring `R` if there is a scalar multiplication `•` (`has_scalar.smul`) that satisfies the expected distributivity axioms for `+` (in `M` and `R`) and `*` (in `R`).
+A module (typeclass `module`) is a semimodule that additionally requires that `R` is a ring and `M` is a group.
+A vector space (typeclass `vector_space`) is a module that additionally requires that `R` is a field.
+All vector spaces are also modules, and all modules are also semimodules.
```

Comment:
Maybe you should emphasize that `M` should already be an `add_comm_monoid` or `add_comm_group`, since this is pretty surprising (and wasn't always like that).